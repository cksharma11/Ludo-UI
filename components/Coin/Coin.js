import React from 'react';
import PropTypes from 'prop-types';
import CoinStyles from './Coin.style';

const Coin = ({ color }) => {
  return (
    <div className={`coin dark_${color}`}>
      <style jsx>{CoinStyles}</style>
    </div>
  );
};

Coin.defaultProps = {
  color: ''
};

Coin.propTypes = {
  color: PropTypes.string
};

export default Coin;
