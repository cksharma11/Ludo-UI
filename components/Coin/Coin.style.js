import css from 'styled-jsx/css';
import colors from '../../config/styles/colors';

const CoinStyles = css`
  .coin {
    width: 80%;
    height: 80%;
    border-radius: 50%;
    margin: 10%;
    border: 1px solid ${colors.gray.shade3};
  }

  .dark_red {
    background: red;
  }

  .dark_green {
    background: green;
  }

  .dark_yellow {
    background: yellow;
  }

  .dark_blue {
    background: blue;
  }
`;

export default CoinStyles;
