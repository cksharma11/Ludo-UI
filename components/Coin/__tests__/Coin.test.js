import React from 'react';
import { shallow } from 'enzyme';
import Coin from '../Coin';

describe('Coin', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Coin />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
