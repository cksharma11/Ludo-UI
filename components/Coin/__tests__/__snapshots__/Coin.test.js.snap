// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin should render properly 1`] = `
"<div className=\\"jsx-2045488686 coin dark_\\">
  <JSXStyle id=\\"2045488686\\">
    .
    c
    o
    i
    n
    .
    j
    s
    x
    -
    2
    0
    4
    5
    4
    8
    8
    6
    8
    6
    {
    w
    i
    d
    t
    h
    :
    8
    0
    %
    ;
    h
    e
    i
    g
    h
    t
    :
    8
    0
    %
    ;
    b
    o
    r
    d
    e
    r
    -
    r
    a
    d
    i
    u
    s
    :
    5
    0
    %
    ;
    m
    a
    r
    g
    i
    n
    :
    1
    0
    %
    ;
    b
    o
    r
    d
    e
    r
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    e
    e
    e
    e
    e
    e
    ;
    }
    .
    d
    a
    r
    k
    _
    r
    e
    d
    .
    j
    s
    x
    -
    2
    0
    4
    5
    4
    8
    8
    6
    8
    6
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    e
    d
    ;
    }
    .
    d
    a
    r
    k
    _
    g
    r
    e
    e
    n
    .
    j
    s
    x
    -
    2
    0
    4
    5
    4
    8
    8
    6
    8
    6
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    g
    r
    e
    e
    n
    ;
    }
    .
    d
    a
    r
    k
    _
    y
    e
    l
    l
    o
    w
    .
    j
    s
    x
    -
    2
    0
    4
    5
    4
    8
    8
    6
    8
    6
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    y
    e
    l
    l
    o
    w
    ;
    }
    .
    d
    a
    r
    k
    _
    b
    l
    u
    e
    .
    j
    s
    x
    -
    2
    0
    4
    5
    4
    8
    8
    6
    8
    6
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    b
    l
    u
    e
    ;
    }
  </JSXStyle>
</div>"
`;
